### PR TITLE
[ONNX] Upgrade exported ONNX IR version to 6

### PR DESCRIPTION
**Pull request by BowenBao**
Upgrade IR version from 4 to 6, below is change doc from ONNX. The upgrade should be backward compatible.

```
  // IR VERSION 5 published on March 18, 2019
  // - Add message TensorAnnotation.
  // - Add quantization annotation in GraphProto to map tensor with its scale and zero point quantization parameters.
  IR_VERSION_2019_3_18 = 0x0000000000000005;

  // IR VERSION 6 published on Sep 19, 2019
  // - Add support for sparse tensor constants stored in model.
  //   - Add message SparseTensorProto
  //   - Add sparse initializers
  IR_VERSION = 0x0000000000000006;
```

### CODE REVIEW

**Review by houseroad — COMMENTED** [review]
Could you rebase to master and rerun the tests?

File: torch/onnx/__init__.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 # TODO: Update these variables when there
 # is a new ir_version and producer_version
 # and use these values in the exporter
-ir_version = 4
+ir_version = 6
 producer_name = "pytorch"
 producer_version = "1.3"
```
Comment (spandantiwari) [inline torch/onnx/__init__.py]: @BowenBao - IR version 6 is only supported in ONNX 1.5 which is opset 11. We are releasing Opset 11 support in the upcoming PyTorch 1.4. Should we update the `producer_version` to 1.4 as well.

File: torch/csrc/jit/export.cpp [hunk 2]
```diff
@@ -217,7 +217,7 @@ EncoderBase::EncoderBase(
   // we pin IR version to version 4 (01/22/2019) instead of using
   // onnx::IR_VERSION. with this change, the test_operators.py will be more
   // stable. only bump it when it's necessary
-  model_proto_.set_ir_version(4);
+  model_proto_.set_ir_version(6);
   // TODO: set the producer version using appropriate function call
   model_proto_.set_producer_version("1.3");
```
Comment (spandantiwari) [inline torch/csrc/jit/export.cpp]: Same comment about `producer_version` here.

**Review by spandantiwari — CHANGES_REQUESTED** [review]

File: torch/csrc/jit/export.cpp (hunk 2 above)
Comment (houseroad) [reply torch/csrc/jit/export.cpp]: We should really unify two variables in py and c++ together :-)

File: torch/onnx/__init__.py (hunk 1 above)
Comment (BowenBao) [reply torch/onnx/__init__.py]: updated

File: torch/csrc/jit/export.cpp (hunk 2 above)
Comment (BowenBao) [reply torch/csrc/jit/export.cpp]: updated, also removed duplication by creating static info in torch/csrc/onnx/onnx.h

**Comment by BowenBao** [comment]
clang-tidy fails with 
```
RuntimeError: Error executing clang-tidy -p build -config {"Checks": "-*, bugprone-*, -bugprone-forward-declaration-namespace, -bugprone-macro-parentheses, -bugprone-lambda-function-name, cppcoreguidelines-*, -cppcoreguidelines-interfaces-global-init, -cppcoreguidelines-owning-memory, -cppcoreguidelines-pro-bounds-array-to-pointer-decay, -cppcoreguidelines-pro-bounds-constant-array-index, -cppcoreguidelines-pro-bounds-pointer-arithmetic, -cppcoreguidelines-pro-type-cstyle-cast, -cppcoreguidelines-pro-type-reinterpret-cast, -cppcoreguidelines-pro-type-static-cast-downcast, -cppcoreguidelines-pro-type-union-access, -cppcoreguidelines-pro-type-vararg, -cppcoreguidelines-special-member-functions, hicpp-exception-baseclass, hicpp-avoid-goto, modernize-*, -modernize-return-braced-init-list, -modernize-use-auto, -modernize-use-default-member-init, -modernize-use-using, performance-*, -performance-noexcept-move-constructor, ", "HeaderFilterRegex": "torch/csrc/.*", "AnalyzeTemporaryDtors": false, "CheckOptions": null} -line-filter [{"name": "torch/csrc/onnx/init.cpp", "lines": [[33, 36]]}] torch/csrc/onnx/init.cpp: /home/runner/work/pytorch/pytorch/third_party/onnx/onnx/onnx_pb.h:50:10: error: 'onnx/onnx-ml.pb.h' file not found [clang-diagnostic-error]
#include "onnx/onnx-ml.pb.h"
```
doesn't seem to relate to this PR.

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit 815474c:
* **2/2** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=815474c084641b95b8576f24fb8a96004148e7f6).

### 2 new failures recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/3930186) caffe2_onnx_py3_6_clang7_ubuntu16_04_test (1/2)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=73565305))
```
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_tan PASSED              [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_topk PASSED             [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_topk_smallest_unsorted PASSED [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_transpose PASSED        [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_type_as PASSED          [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_unfold PASSED           [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_unique PASSED           [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_unsqueeze PASSED        [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_upsample_nearest_scale FAILED [  4%] 
Dec 12 21:31:35 test/onnx/test_operators.py::TestOperators::test_upsample_nearest_scale_default_scale_factor PASSED [  4%] 
Dec 12 21:31:36 test/onnx/test_operators.py::TestOperators::test_upsample_nearest_size FAILED [  5%] 
Dec 12 21:31:36 test/onnx/test_operators.py::TestOperators::test_view PASSED             [  5%] 
Dec 12 21:31:36 test/onnx/test_operators.py::TestOperators::test_view_flatten PASSED     [  5%] 
Dec 12 21:31:36 test/onnx/test_pytorch_helper.py::TestCaffe2Backend::test_helper SKIPPED [  5%] 
Dec 12 21:31:36 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test__dim_arange PASSED [  5%] 
Dec 12 21:31:36 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test__sample_dirichlet PASSED [  5%] 
Dec 12 21:31:36 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test__standard_gamma PASSED [  5%] 
Dec 12 21:31:36 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test_adaptive_avg_pool1D PASSED [  5%] 
Dec 12 21:31:36 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test_adaptive_avg_pool2D PASSED [  5%] 
Dec 12 21:31:36 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test_adaptive_avg_pool3D PASSED [  5%] 
Dec 12 21:31:37 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test_adaptive_max_pool1D PASSED [  5%] 
```
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/3930194) caffe2_onnx_py2_gcc5_ubuntu16_04_test (2/2)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=73565448))
```
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_tan PASSED              [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_topk PASSED             [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_topk_smallest_unsorted PASSED [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_transpose PASSED        [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_type_as PASSED          [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_unfold PASSED           [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_unique PASSED           [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_unsqueeze PASSED        [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_upsample_nearest_scale FAILED [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_upsample_nearest_scale_default_scale_factor PASSED [  4%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_upsample_nearest_size FAILED [  5%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_view PASSED             [  5%] 
Dec 12 21:32:20 test/onnx/test_operators.py::TestOperators::test_view_flatten PASSED     [  5%] 
Dec 12 21:32:20 test/onnx/test_pytorch_helper.py::TestCaffe2Backend::test_helper SKIPPED [  5%] 
Dec 12 21:32:20 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test__dim_arange PASSED [  5%] 
Dec 12 21:32:20 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test__sample_dirichlet PASSED [  5%] 
Dec 12 21:32:20 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test__standard_gamma PASSED [  5%] 
Dec 12 21:32:20 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test_adaptive_avg_pool1D PASSED [  5%] 
Dec 12 21:32:20 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test_adaptive_avg_pool2D PASSED [  5%] 
Dec 12 21:32:21 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test_adaptive_avg_pool3D PASSED [  5%] 
Dec 12 21:32:21 test/onnx/test_pytorch_onnx_caffe2.py::TestCaffe2BackendEmbed_opset9::test_adaptive_max_pool1D PASSED [  5%] 
```


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

**Review by spandantiwari — APPROVED** [review]
Thanks for the changes. LGTM.

File: torch/onnx/__init__.py [hunk 3]
```diff
@@ -9,9 +9,9 @@
 # TODO: Update these variables when there
 # is a new ir_version and producer_version
 # and use these values in the exporter
-ir_version = 4
+ir_version = _C._onnx.IR_VERSION
```
Comment (houseroad) [inline torch/onnx/__init__.py]: Thanks for merging python and c++ variables

**Review by houseroad — APPROVED** [review]
Looks good.

**Review by houseroad — COMMENTED** [review]
Actually landing this PR is tricky. Since we merged some other ONNX related PRs earlier, which may introduce new expect files. Could you rebase to master to make sure we are 100% safe?

Thanks.

**Comment by BowenBao** [comment]
> Actually landing this PR is tricky. Since we merged some other ONNX related PRs earlier, which may introduce new expect files. Could you rebase to master to make sure we are 100% safe?
> 
> Thanks.

Rebased just now.

**Review by houseroad — CHANGES_REQUESTED** [review]
The base branch needs to be  v1.4.0

**Review by houseroad — APPROVED** [review]

**Comment by houseroad** [comment]
This is not a cherrypick, my bad

**Pull request closed without merging**